commands/remove.rb

```diff
@@ -13,10 +14,20 @@ def self.remove(pkg, verbose)
       return
     end
 
-    # Don't remove any of the packages ruby (and thus crew) needs to run.
-    if CREW_ESSENTIAL_PACKAGES.include?(pkg.name)
-      puts "Refusing to remove essential package #{pkg.name}.".lightred
-      return
+    # Determine dependencies of packages in CREW_ESSENTIAL_PACKAGES and
+    # their dependencies, as those are needed for ruby and crew to run,
+    # and thus should not be removed.
+    essential_deps = CREW_ESSENTIAL_PACKAGES.flat_map { |essential_pkg| Package.load_package("#{essential_pkg}.rb").get_deps_list }.push(*CREW_ESSENTIAL_PACKAGES).uniq.sort
+
+    if essential_deps.include?(pkg.name)
+      return if Package.load_package("#{pkg.name}.rb").in_upgrade
+
+      puts <<~ESSENTIAL_PACKAGE_WARNING_EOF.gsub(/^(?=\w)/, '  ').lightred
+        #{pkg.name.capitalize} is considered an essential package needed for
+        Chromebrew to function and thus cannot be removed.
+      ESSENTIAL_PACKAGE_WARNING_EOF
```

Comment:
I'm confused-- why can't we just use one or to `puts` lines here?

Reply:
It looks really nice!

Reply:
The doc block method is easier to read and adjust the text, if necessary.